concept/answer/Numeric.java

```diff
@@ -0,0 +1,71 @@
+package grakn.client.concept.answer;
+
+import grakn.client.common.exception.GraknClientException;
+import grakn.protocol.AnswerProto;
+
+import javax.annotation.Nullable;
+
+public class Numeric {
```

Comment:
I think it would be more consistent with the rest of our codebase if we split up `Numeric` into 3 inner classes: `Numeric.Long`, `Numeric.Double` and `Numeric.NaN`.

Reply:
And then have `asLong()` and `asDouble()` return `Numeric.Long` and `Numeric.Double` rather than `long` and `double`?

I can see how that's how we're doing it for attribute type, for example, so I can see the consistency. However, it'll make the code more verbose.

What do you guys think @haikalpribadi @alexjpwalker ?

Reply:
> And then have `asLong()` and `asDouble()` return `Numeric.Long` and `Numeric.Double` rather than `long` and `double`?

Yes and I agree with your thoughts too - let's see what Haikal thinks

Reply:
I think it would be too verbose, @alexjpwalker @lolski . I think the user should get the `primitive types` when they call `.longValue()` and `.doubleValue()`.

Reply:
and `NaN` does not even need a class anyways, right.

Reply:
ps: `NaN` goes well with curry :D ugh I'm still hungry

Reply:
haha mark as spam